reference_validator/reference_validator.py

```diff
@@ -1,786 +1,42 @@
-                        for prop, value in res.properties.iteritems():
-                            if value == False:
-                                if self.pretty_format:
-                                    print('- ' + YAML_colours.YELLOW + prop + YAML_colours.DEFAULT +
-                                          ' in ' + YAML_colours.YELLOW + res.name +
-                                          YAML_colours.DEFAULT)
-                                else:
-                                    print('- ' + prop + ' in ' + res.name)
-                    print('')
-
-                # Print unused resources (optional)
-                if (self.print_unused) and [True for x in node.resources if not x.used]:
-                    if (self.pretty_format):
-                        print(YAML_colours.BOLD + 'Resources without reference:' +
-                              YAML_colours.DEFAULT)
-                    else:
-                        print('Resources without reference:')
-
-                    for resource in node.resources:
-                        if resource.used == False:
-                            if self.pretty_format:
-                                print('- ' + YAML_colours.YELLOW + resource.name + YAML_colours.DEFAULT +
-                                      ' (' + resource.type + ')')
-                            else:
-                                print('- ' + resource.name)
-                    print('')
-
-                # Print file status as OK if there were no problems
-                if node.ok:
-                    if self.pretty_format:
-                        print(YAML_colours.BOLD + 'Status: ' + YAML_colours.GREEN + 'OK' +
-                              YAML_colours.DEFAULT)
-                    else:
-                        print ('Status: OK')
-                else:
-                    if self.pretty_format:
-                        print(YAML_colours.BOLD + 'Status: ' + YAML_colours.RED + 'FAILED' +
-                              YAML_colours.DEFAULT)
-                    else:
-                        print('Status: FAILED')
-
-                print('\n\n')
 
+import YAML_HotValidator as VALIDATOR
 
 def main():
-    
+
     # Parse arguments
     parser = argparse.ArgumentParser()
-    parser.add_argument('-u', '--unused', action='store_true',
+    parser.add_argument('-u', '--print-unused', action='store_true',
                         help='When true, prints all unused resources/parameters.')
     parser.add_argument('-p', '--pretty-format', action='store_true',
                         help='When true, provides colourful output')
-    parser.add_argument('-e', '--environment', metavar='path/to/environment', nargs='+',
+    parser.add_argument('-t', '--print-tree', action='store_true',
+                        help='When true, output contains template structure')
+    parser.add_argument('-e', '--environment-file', metavar='path/to/environment', nargs='+',
```

Comment:
This should say `action='append'` instead of `nargs` here.

`nargs='+'` does `-e file1.yaml file2.yaml file3.yaml`, but Heat (and therefore this) expects: `-e file1.yaml -e file2.yaml -e file3.yaml` which is what action=append does.